lib/smf/smf.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2021 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "smf.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(smf);
+
+/*
+ * Private structure (to this file) used to track state machine context.
+ * The structure is not used directly, but instead to cast the "internal"
+ * member of the smf_ctx structure.
+ */
+struct internal_ctx {
+	const struct smf_state *last_entered;
+	uint32_t running : 1;
+	uint32_t enter	 : 1;
+	uint32_t exit	 : 1;
```

Comment:
since they seem to be boolean flags, `bool running: 1` could be used

Reply:
Done